src/js/constants.js

```diff
@@ -1,22 +1,52 @@
-import {analog, obisObserver, mtx, utils} from 'jooby-codec';
-import {directions} from 'jooby-codec/constants/index.js';
+import * as joobyCodec from 'jooby-codec';
 
 
-const {DOWNLINK, UPLINK} = directions;
+const prepareCommandMap = ( commandMap, direction ) => {
+    Object.keys(commandMap).forEach(commandName => {
+        commandMap[commandName].directionType = direction;
+        commandMap[commandName].hasParameters = Object
+            .values(commandMap[commandName].examples)
+            .some(example => Object.keys(example.parameters).length > 0);
+    });
 
-export const directionNames = {
-    [DOWNLINK]: 'downlink',
-    [UPLINK]: 'uplink'
+    return commandMap;
 };
 
-const prepareCommandList = commands => [...Object.values(commands.uplink), ...Object.values(commands.downlink)]
-    .filter(item => item.id)
+const prepareCommandList = CommandMap => [
+    ...Object.values(CommandMap.uplink),
+    ...Object.values(CommandMap.downlink)
+]
     .map(item => ({
-        label: item.name,
         value: item,
+        label: item.name,
         direction: directionNames[item.directionType]
     }));
 
+export const directions = {
+    DOWNLINK: 1,
+    UPLINK: 2
+};
+
+export const directionNames = {
+    [directions.DOWNLINK]: 'downlink',
+    [directions.UPLINK]: 'uplink'
+};
+
+export const commands = {
```

Comment:
is it really a constant?

Reply:
Added the file `joobyCodec.js` to the root of the project.